### DISCUSSION TITLE
Property assignment does not work with classmethod in MicroPython?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16344] [General]
Property assignment does not work with classmethod in MicroPython?
```
class MyClass:
    _my_static_property = 10

    @property
    @classmethod
    def my_static_property(cls):
        return cls._my_static_property

# Accessing the property
print(MyClass.my_static_property)
'<property>'
```
Regular python I would get the actual value printed, i.e. 10.  If I flip the attributes I get "'\<bound method\>'", so appears the parser only honors the first property found?  As I recall Python 3.11 now allows multiple attributes or directives assignment.   https://docs.python.org/3.11/library/functions.html#classmethod https://docs.python.org/3.11/library/functions.html#propery

**Comment by Josverl** [comment]
Micropython is based off the Python 3.4 specs , with several features and functionality from newer CPython versions as documented in **https://docs.micropython.org/en/latest/genrst/index.html** 
It is not correct to assume that all features from 3.11 are implemented.

**Comment by sosi-deadeye** [comment]
https://docs.python.org/3.11/whatsnew/3.11.html#language-builtins

> Chaining [classmethod](https://docs.python.org/3.11/library/functions.html#classmethod) descriptors (introduced in [bpo-19072](https://bugs.python.org/issue?@action=redirect&bpo=19072)) is now deprecated. It can no longer be used to wrap other descriptors such as [property](https://docs.python.org/3.11/library/functions.html#property). The core design of this feature was flawed and caused a number of downstream problems. To “pass-through” a [classmethod](https://docs.python.org/3.11/library/functions.html#classmethod), consider using the __wrapped__ attribute that was added in Python 3.10. (Contributed by Raymond Hettinger in [gh-89519](https://github.com/python/cpython/issues/89519).)

**Comment by Jibun-no-Kage** [comment]
Yeah, was not assuming just asking, if MicroPython did it.  A clarification question.  In my use case would have been of benefit... easy to leverage.  As to 'wrapped' attribute thank you, excellent referencing.  I prefer to use properties when things are 'read' only if you well, it communicates to the user that this method is 'read' only.  Ignoring the 'setter' versus a simple function declaration, but I digress.  Very interesting reading about the issue.  First came across 'wrappers' in passing arguments to Timers... in MicroPython.
